### PR TITLE
Speed up percentile standard error calculation

### LINKED ISSUE #1772: [TASK] Improve performance of percentile standard error calculation
--------
Issue Description
--------

**Is your feature request related to a problem? Please describe.**
The current implementation of percentile standard error when a pivot parameter is present loops over target, then percent, and finally over pivot parameter values. A KDE is built each time in the innermost loop. This could be made more efficient.

**Describe the solution you'd like**
Changing the order of the loops would reduce the number of times the KDE is built. Looping over target, then pivot parameter values, and finally over percent would do this since the data associated with the pivot parameter does not change based on the requested percentile.

**Describe alternatives you've considered**
None

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or task?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest release branch? If yes, is there any issue tagged with release (create if needed)?
- [x] 5. If the issue is being closed without a pull request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by dgarrett622**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #1772 

##### What are the significant changes in functionality due to this change request?
Changes order of loops to minimize number of times KDEs are built to speed up calculation of percentile standard error.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

File: framework/Models/PostProcessors/BasicStatistics.py [hunk 1]
```diff
@@ -999,20 +999,21 @@ def __runLocal(self, inputData):
         targDa = dataSet[target]
         if self.pivotParameter in targDa.sizes.keys():
           percentileSte = []
-          for pct in percent:
+          for label, group in targDa.groupby(self.pivotParameter):
```
Comment (wangcj05) [inline framework/Models/PostProcessors/BasicStatistics.py:1002]: These changes will decrease the running time for the following test from 480s to 335s. 
image

The original test before the add of error calculation for percentile is about 130s.

**Comment by wangcj05** [comment]
@dgarrett622 Do you to include the modification that we discussed today in this PR or in a separate PR?

File: framework/Models/PostProcessors/BasicStatistics.py (hunk 1 above)
Comment (PaulTalbot-INL) [reply framework/Models/PostProcessors/BasicStatistics.py:1002]: Have we run any tests on the accuracy of the "standard normal" assumption versus this fitted function? I'm sure it's heavily reliant on the particular model being sampled, but some idea of the error introduced would inform how valuable the more accurate calculation is.

**Comment by PaulTalbot-INL** [comment]
Since this is passing tests and the work is done, I can review it now, and an issue can be opened for the standard normal calculation as a speedup option, along with determining which should be default.

**Review by PaulTalbot-INL — APPROVED** [review]

**Comment by PaulTalbot-INL** [comment]
Checklists pass.

@dgarrett622 please open an issue to track @wangcj05 comments regarding speedup using the standard normal error assumption.

**Pull request merged**